packages/medusa/src/services/event-bus.ts

```diff
@@ -99,22 +98,52 @@ export default class EventBusService {
     const cloned = new EventBusService(
       {
         manager: transactionManager,
```

Comment:
suggestion: This method can return this directly and noted as deprecated

Reply:
Nice catch!